torch/_dynamo/variables/builder.py

```diff
@@ -522,11 +522,12 @@ def index_source(key):
             )
         elif (
             istype(value, (type, types.FunctionType))
-            and skipfiles.check(getfile(value), allow_torch=True)
+            and skipfiles.check(getfile(value), allow_torch=True).skipped
             and not inspect.getattr_static(value, "_torchdynamo_inline", False)
         ):
             return SkipFilesVariable(
                 value,
+                skipfiles.check(getfile(value)).reason,
```

Comment:
it's kind of tiresome that you have to call skipfiles.check twice here

Reply:
[````Assignment Expressions```](https://peps.python.org/pep-0572/) can help us, but it's available since Python 3.8, we still support 3.7.